### PR TITLE
Expose streams in all public sorting APIs

### PR DESCRIPTION
**Pull request by vyasr**
## Description

Contributes to #925

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

**Review by bdice — APPROVED** [review]
Some comments need to be reflowed, but otherwise LGTM.

File: cpp/include/cudf/sorting.hpp [hunk 1]
```diff
@@ -65,27 +68,30 @@ std::unique_ptr<column> stable_sorted_order(
...
  * @brief Checks whether the rows of a `table` are sorted in a lexicographical
  *        order.
  *
- * @param[in] table             Table whose rows need to be compared for ordering
- * @param[in] column_order      The expected sort order for each column. Size
+ * @param table             Table whose rows need to be compared for ordering
+ * @param column_order      The expected sort order for each column. Size
  *                              must be equal to `in.num_columns()` or empty. If
  *                              empty, it is expected all columns are in
  *                              ascending order.
```
[the remark this answers is not in the document]
Comment (vyasr) [reply cpp/include/cudf/sorting.hpp:82]: Ugh thanks yeah I'll fix.

File: cpp/include/cudf/sorting.hpp [hunk 2]
```diff
@@ -65,27 +68,30 @@ std::unique_ptr<column> stable_sorted_order(
...
+ *                          must be equal to `in.num_columns()` or empty. If
+ *                          empty, it is expected all columns are in
+ *                          ascending order.
+ * @param null_precedence   The desired order of null compared to other
+ *                          elements for each column. Size must be equal to
+ *                          `input.num_columns()` or empty. If empty,
+ *                          `null_order::BEFORE` is assumed for all columns.
+ *
+ * @param stream CUDA stream used for device memory operations and kernel launches
  * @returns bool                true if sorted as expected, false if not
```
Comment (davidwendt) [inline cpp/include/cudf/sorting.hpp:89]:
```suggestion
 * @returns true if sorted as expected, false if not
```
Don't include the type in the `@return` tag

**Review by davidwendt — APPROVED** [review]

**Pull request merged**